earth_enterprise/rpms/post-install_fusion.sh

```diff
@@ -0,0 +1,303 @@
+        echo ""
+    fi
+}
+
+check_fusion_master_or_slave()
+{
+    if [ -f "$ASSET_ROOT/.config/volumes.xml" ]; then
+        EXISTING_HOST=$(xmllint --xpath "//VolumeDefList/volumedefs/item[1]/host/text()" $ASSET_ROOT/.config/volumes.xml | $NEWLINECLEANER)
+
+        case "$EXISTING_HOST" in
+			$HOSTNAME_F|$HOSTNAME_A|$HOSTNAME_S|$HOSTNAME)
+                IS_SLAVE=false
+                ;;
+            *)
+                IS_SLAVE=true
+
+                echo -e "\nThe asset root [$ASSET_ROOT] is owned by another Fusion host:  $EXISTING_HOST"
+                echo -e "Installing $GEEF $LONG_VERSION in Grid Slave mode.\n"
+                ;;
+        esac
+    fi
+}
+
+create_systemrc()
+{
+	echo "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\" ?>
+<Systemrc>
+  <assetroot>$ASSET_ROOT</assetroot>
+  <maxjobs>$NUM_CPUS</maxjobs>
+  <locked>0</locked>
+  <fusionUsername>$GEFUSIONUSER_NAME</fusionUsername>
+  <userGroupname>$GROUPNAME</userGroupname>
+</Systemrc>" > $SYSTEMRC
+}
+
+install_or_upgrade_asset_root()
+{
+    if [ -f "$SYSTEMRC" ]; then
+        rm -f $SYSTEMRC
+    fi
+
+    # create new systemrc file
+    create_systemrc
+
+    chmod 644 $SYSTEMRC
+    chown $GEFUSIONUSER_NAME:$GROUPNAME $SYSTEMRC
+
+    if [ $IS_NEWINSTALL == true ]; then
+        $BASEINSTALLDIR_OPT/bin/geconfigureassetroot --new --noprompt --assetroot $ASSET_ROOT --srcvol $SOURCE_VOLUME
+        chown -R $GEFUSIONUSER_NAME:$GROUPNAME $ASSET_ROOT
+    else
+        # upgrade asset root -- if this is a master
+        if [ $IS_SLAVE == false ]; then
+            # TODO: Verify this logic -- this is what is defined in the installer documentation, but need confirmation            
+            if [ $NEW_GEGROUP == true ] || [ $NEW_GEFUSIONUSER == true ]; then
+                NOCHOWN=""
+                UPGRADE_MESSAGE="\nThe upgrade will fix permissions for the asset root and source volume. This may take a while.\n"
+            else
+                NOCHOWN="--nochown"
+                UPGRADE_MESSAGE=""
+            fi
+
+            echo -e "\nThe asset root must be upgraded to work with the current version of $GEEF $LONG_VERSION."
+            echo -e "You cannot use an upgraded asset root with older versions of $GEEF. "
+	    echo -e "Consider backing up your asset root. $GEEF will warn you when"
+            echo -e "attempting to run with a non-upgraded asset root."
+            echo -e "$UPGRADE_MESSAGE"
+                
+		# Note: we don't want to do the recursive chown on the asset root unless absolutely necessary
+            $BASEINSTALLDIR_OPT/bin/geconfigureassetroot --fixmasterhost --noprompt  $NOCHOWN --assetroot $ASSET_ROOT
+            $BASEINSTALLDIR_OPT/bin/geupgradeassetroot --noprompt $NOCHOWN --assetroot $ASSET_ROOT
+	    chown -R $GEFUSIONUSER_NAME:$GROUPNAME $ASSET_ROOT
+        fi  
+    fi
+}
+
+fix_postinstall_filepermissions()
```

Comment:
This should be done while files are being moved into the package, not in a post-install script.

Reply:
ok